ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -361,7 +398,10 @@ let assert_gpus_available ~__context ~self ~host =
     raise (Api_errors.Server_error (Api_errors.vm_requires_gpu, [
         Ref.string_of self;
         Ref.string_of (List.hd not_available)
-      ]))
+      ]));
+  (* If the pgpus are available, check also vgpu/pgpu compatibility *)
+  assert_gpus_compatible ~__context ~self ~host
```

Comment:
The main issue I have is that this function, as part of `assert_can_boot_here`, is not just called on migration, by also on start and resume. The compatibility checks, however, are strictly for running VMs/vGPUs. So I think that this is the wrong place to do this.